src/jit/SimdArm32Inl.h

```diff
@@ -300,25 +620,167 @@ static void simdEmitI32x4Dot(sljit_compiler* compiler, sljit_s32 vd, sljit_s32 v
+        ASSERT(SLJIT_IS_MEM(src.arg));
+        sljit_emit_fmem(compiler, SLJIT_MOV_F64 | SLJIT_MEM_UNALIGNED | SLJIT_MEM_LOAD, SLJIT_FR0, src.arg, srcMem);
+        sljit_emit_fop1(compiler, op, dst.arg, dstMem, SLJIT_FR0, 0);
+    }
+}
+
+static void simdEmitF64x2Unary(sljit_compiler* compiler, JITArg src, JITArg dst, uint32_t op)
+{
+    auto srcMem = src.argw;
+    auto dstMem = dst.argw;
+
+    for (uint8_t i = 0; i < 2; i++, srcMem += sizeof(double), dstMem += sizeof(double)) {
+        ASSERT(SLJIT_IS_MEM(src.arg));
+        sljit_emit_fmem(compiler, SLJIT_MOV_F64 | SLJIT_MEM_UNALIGNED | SLJIT_MEM_LOAD, SLJIT_FR0, src.arg, srcMem);
+        sljit_emit_op_custom(compiler, &op, sizeof(op));
+        ASSERT(SLJIT_IS_MEM(dst.arg));
+        sljit_emit_fmem(compiler, SLJIT_MOV_F64 | SLJIT_MEM_UNALIGNED | SLJIT_MEM_STORE, SLJIT_FR0, dst.arg, dstMem);
+    }
+}
+
+static void simdEmitF64x2Binary(sljit_compiler* compiler, JITArg src[2], JITArg dst, sljit_s32 op)
+{
+    auto src0Mem = src[0].argw;
+    auto src1Mem = src[1].argw;
+    auto dstMem = dst.argw;
+
+    for (uint8_t i = 0; i < 2; i++, src0Mem += sizeof(double), src1Mem += sizeof(double), dstMem += sizeof(double)) {
+        ASSERT(SLJIT_IS_MEM(src[0].arg));
+        ASSERT(SLJIT_IS_MEM(src[1].arg));
+        sljit_emit_fmem(compiler, SLJIT_MOV_F64 | SLJIT_MEM_UNALIGNED | SLJIT_MEM_LOAD, SLJIT_FR0, src[0].arg, src0Mem);
+        sljit_emit_fmem(compiler, SLJIT_MOV_F64 | SLJIT_MEM_UNALIGNED | SLJIT_MEM_LOAD, SLJIT_FR1, src[1].arg, src1Mem);
+        sljit_emit_fop2(compiler, op, dst.arg, dstMem, SLJIT_FR0, 0, SLJIT_FR1, 0);
+    }
+}
+
+static void simdEmitF64x2CMP(sljit_compiler* compiler, JITArg src[2], JITArg dst, sljit_s32 opcode, sljit_s32 flag)
+{
+    opcode |= SLJIT_CMP_F64;
+    auto tmpReg = SLJIT_R0;
+    auto src0Mem = src[0].argw;
+    auto src1Mem = src[1].argw;
+
+    for (uint8_t i = 0; i < 2; i++, src0Mem += sizeof(double), src1Mem += sizeof(double)) {
+        sljit_emit_fmem(compiler, SLJIT_MOV_F64 | SLJIT_MEM_LOAD | SLJIT_MEM_UNALIGNED, SLJIT_FR0, src[0].arg, src0Mem);
+        sljit_emit_fmem(compiler, SLJIT_MOV_F64 | SLJIT_MEM_LOAD | SLJIT_MEM_UNALIGNED, SLJIT_FR1, src[1].arg, src1Mem);
+        sljit_emit_fop1(compiler, opcode, SLJIT_FR0, 0, SLJIT_FR1, 0);
+        sljit_emit_op_flags(compiler, SLJIT_MOV, tmpReg, 0, flag);
+        sljit_emit_op2(compiler, SLJIT_SUB, tmpReg, 0, SLJIT_IMM, 0, tmpReg, 0);
+        sljit_emit_mem(compiler, SLJIT_MOV | SLJIT_MEM_STORE | SLJIT_MEM_UNALIGNED, tmpReg, dst.arg, dst.argw + ((i * 2) * sizeof(float)));
+        sljit_emit_mem(compiler, SLJIT_MOV | SLJIT_MEM_STORE | SLJIT_MEM_UNALIGNED, tmpReg, dst.arg, dst.argw + (((i * 2) + 1) * sizeof(float)));
+    }
+}
+
 static void emitUnarySIMD(sljit_compiler* compiler, Instruction* instr)
 {
     Operand* operands = instr->operands();
     JITArg args[2];
 
     sljit_s32 type = SLJIT_SIMD_ELEM_128;
     bool isDSTNormalRegister = false;
+    bool isCallback = true;
```

Comment:
This should be false by default.

Reply:
Callback is more times used. I think it is better to set this to true in the beginning.